injection/sharedmain/main.go

```diff
@@ -63,18 +64,39 @@ func GetConfig(masterURL, kubeconfig string) (*rest.Config, error) {
 	if kubeconfig == "" {
 		kubeconfig = os.Getenv("KUBECONFIG")
 	}
+
+	// We produce configs a bunch of ways, this gives us a single place
+	// to "decorate" them with common useful things (e.g. for debugging)
+	decorate := func(cfg *rest.Config) *rest.Config {
+
+		// Augment the rest.Config with a "wrapper" around the transport that
+		// will emit varying levels of debug logging when -v is passed with
+		// levels 6 to 9.
+		wt := transport.DebugWrappers
+		if cfg.WrapTransport != nil {
+			wt = transport.Wrappers(wt, cfg.WrapTransport)
+		}
+		cfg.WrapTransport = wt
+
+		return cfg
+	}
+
 	// If we have an explicit indication of where the kubernetes config lives, read that.
 	if kubeconfig != "" {
-		return clientcmd.BuildConfigFromFlags(masterURL, kubeconfig)
+		if c, err := clientcmd.BuildConfigFromFlags(masterURL, kubeconfig); err != nil {
+			return nil, err
+		} else {
+			return decorate(c), nil
+		}
```

Comment:
unindent

Reply:
then I need to unbundle the declaration so that `c` is in scope....